acvm-repo/brillig/src/opcodes.rs

```diff
@@ -301,20 +316,26 @@ pub enum BrilligOpcode<F> {
         source_b: MemoryAddress,
         condition: MemoryAddress,
     },
+    /// Reads the `source_pointer` to obtain a memory address, then retrieves the data
+    /// stored at that address and writes it to the `destination` address.
     Load {
         destination: MemoryAddress,
         source_pointer: MemoryAddress,
     },
+    /// Reads the `destination_pointer` to obtain a memory address, then stores the value
+    /// from the `source` address at that location.
     Store {
         destination_pointer: MemoryAddress,
         source: MemoryAddress,
     },
+    /// Native functions in the VM.
+    /// These are equivalent to the black box functions in ACIR.
     BlackBox(BlackBoxOp),
-    /// Used to denote execution failure, returning data after the offset
+    /// Used to denote execution failure, halting the VM and returning data specified by a memory-sized vector.
```

Comment:
What does it mean to be `memory-sized`? Will this opcode fail if the `HeapVector` is of a different size?

Reply:
Updated to `dynamically-sized`